### PR TITLE
Refactor commands the final

### PR DESCRIPTION
**Pull request by bzeller**
This PR moves the last remaining commands over to the new style command objects. Also removes some now unused code.

### CODE REVIEW

**Comment by mlandres** [comment]
```
-msgid "Print help."
+msgid "Print zypper help"

-msgid "Repository Options:"
+msgid "Repository Options"

-msgid "Target Options:"
+msgid "Target Options"
```
Please add the 3 old strings to `legacy_strings.cc`. I'll fix the translations later.

File: src/commands/configtest.cc [hunk 1]
```diff
@@ -128,15 +113,23 @@ namespace
...
+ConfigTestCmd::ConfigTestCmd(const std::vector<std::string> &commandAliases_r):
+  ZypperBaseCommand (
+    commandAliases_r,
+    // translators: command synopsis; do not translate lowercase words
+    _("configtest"),
+    // translators: command summary: configtest
+    _("This command is for debuging purposes only."),
+    // translators: command description
+    _("This command is for debuging purposes only."),
+    DisableAll
```
[the remark this answers is not in the document]
Comment (bzeller) [reply src/commands/configtest.cc]: Done

File: src/commands/subcommand.cc [hunk 2]
```diff
@@ -476,65 +476,23 @@ std::ostream & SubcommandOptions::showHelpOn( std::ostream & out ) const
...
-
-int subcommand( Zypper & zypper_r )
-{
-  zypper_r.cleanupForSubcommand();
-  return SubcommandImpl( zypper_r ).run();
-}
-
-int subcommand( Zypper & zypper_r, shared_ptr<SubcommandOptions> options_r )
-{
-  zypper_r.cleanupForSubcommand();
```
Comment (mlandres) [inline src/commands/subcommand.cc]: Looks like the `zypper_r.cleanupForSubcommand();` call is lost (here and from L527). 
Should probably be added as 1st action to `SubCmd::runCmd`.

File: src/Zypper.cc [hunk 3]
```diff
@@ -486,9 +230,15 @@ int Zypper::main( int argc, char ** argv )
...
-	  subcommand( *this );
+  case ZypperCommand::SUBCOMMAND_e: {
+    std::shared_ptr<SubCmd>  cmd = std::dynamic_pointer_cast<SubCmd>( command().commandObject() );
+    if ( cmd )
+      cmd->runCmd( *this );
+    else {
+      ZYPP_THROW( Exception ("Invalid command object") );
+    }
 	  break;
+  }
```
Comment (mlandres) [inline src/Zypper.cc]: Please indent the case block to match the other cases.

**Pull request merged**